### PR TITLE
Tilt Input add-on LS/RS normalisation

### PR DESCRIPTION
**Pull request by InfraredAces**
This PR changes the following things with the Tilt Input add-on

- Make both Right Analog Stick and Left analog stick behave identically to one another compared to the current implementation
- Changes how the tilt factor values entered into the web configurator alter the magnitude of the direction input
- Adds documentation for the new behavior
- Adds input validation for the Tilt Factor Values, limiting them to values 0-100
- Sets default values in add-on for all values, except the Pico config to 0
- Sets default values of Pico Config to match the original values prior to custom values (T1 - 35%, T2 - 65%)

### Left Analog Stick

- Left Analog Stick (LS Tilt 1 Factor XY = 35) - Input Left while holding Tilt 1 => Send input at 65% Left
- Left Analog Stick (LS Tilt 1 Factor XY = 35) - Input Up while holding Tilt 1 => Send input at 65% Up
- Left Analog Stick (LS Tilt 2 Factor XY = 65) - Input Left while holding Tilt 1 => Send input at 35% Left
- Left Analog Stick (LS Tilt 2 Factor XY = 65) - Input Up while holding Tilt 1 => Send input at 35% Up

Pressing tilt with a Tilt Factor Left = N results in a (100-N) so a 35 Tilt Factor value results in a 100-35%=65%.

### Right Analog Stick

The X-axis of the Right analog stick behaves the same as the Left stick, but the Y-axis behaves in a different manner. At 0 Tilt Factor Right Y, RS would go all the way to the top, and at 200, it will go all the way to the bottom. 100 will give you the middle point.

Up and down are entirely unaffected by either T1 or T2 and pressing T1 or T2 + Left or Right will ignore the Up or Down inputs and treat it as if you only input T1 or T2 + Left or Right.

Current Documentation - `The Right stick shifts down for Tilt 1 and up for Tilt 2 by default.`

This behavior **only** occurs with the specific values that are not set as default anywhere other than the Pico config, otherwise the default values are set to -1. Other values will have behavior depended on the actual values used.

#### "Default" Behavior

- Right Analog Stick (RS Tilt 1 Factor X = 35, Tilt Factor 1 Y = 170) - Input only Left while holding Tilt 1 => Send input at 65% Left, 65% Down
- Right Analog Stick (RS Tilt 1 Factor X = 35, Tilt Factor 1 Y = 170) - Input only Right while holding Tilt 1 => Send input at 65% Right, 65% Up
- Right Analog Stick (RS Tilt 2 Factor X = 30, Tilt Factor 2 Y = 30) - Input only Left while holding Tilt 2 => Send input at 70% Left, 70% Up
- Right Analog Stick (RS Tilt 2 Factor X = 30, Tilt Factor 2 Y = 30) - Input only Right while holding Tilt 2 => Send input at 70% Right, 70% Up

#### Behavior with other values

When pressing T1 or T2 + Left or Right with Tilt Factor X RS = N, RS X-axis component behaviors will remain similar to how LS X/Y Axis where it will input (100-N)% of right or left. Y-axis component of input however is determined based on the value map below.

RS Tilt Factor Y-axis Value Map
0 (100% Up)
|
100 (Neutral
|
200 (100% Down)

Up and down are entirely unaffected by either T1 or T2 and pressing T1 or T2 + Left or Right will ignore the Up or Down inputs and treat it as if you only input T1 or T2 + Left or Right.

### Left Analog Stick

Behaves similarly to current implementation. The change is how the Tilt Factor changes the value.

- Left Analog Stick (LS Tilt 1 Factor XY = 35) - Input Left while holding Tilt 1 => Send input at 35% Left
- Left Analog Stick (LS Tilt 1 Factor XY = 35) - Input Up while holding Tilt 1 => Send input at 35% Up
- Left Analog Stick (LS Tilt 2 Factor XY = 65) - Input Left while holding Tilt 1 => Send input at 65% Left
- Left Analog Stick (LS Tilt 2 Factor XY = 65) - Input Up while holding Tilt 1 => Send input at 65% Up

Pressing Tilt 1/2 with a value of N and a direction results in an input that is N% of the maximum direction.

### Right Analog Stick

Behaves identically to the PR Left Analog Stick behavior.

- Right Analog Stick (RS Tilt 1 Factor XY = 35) - Input Left while holding Tilt 1 => Send input at 35% Left
- Right Analog Stick (RS Tilt 1 Factor XY = 35) - Input Up while holding Tilt 1 => Send input at 35% Up
- Right Analog Stick (RS Tilt 2 Factor XY = 65) - Input Left while holding Tilt 1 => Send input at 65% Left
- Right Analog Stick (RS Tilt 2 Factor XY = 65) - Input Up while holding Tilt 1 => Send input at 65% Up

Pressing Tilt 1/2 with a value of N and a direction results in an input that is N% of the maximum direction.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ modules/Custom.cmake
 cmake-*
 env.ini
 
+.docusaurus
```
Comment (arntsonl) [inline .gitignore:17]: Did this sneak in?

**Review by arntsonl — CHANGES_REQUESTED** [review]
Quick change on .gitignore

**Comment by mikepparks** [comment]
Reimplemented this in #585 to clean up some conflicts and include the recent changes to analog centers.

**Comment by InfraredAces** [comment]
Closing PR in favor of #585 as it contains all changes made in this PR with additional reworks

**Pull request closed without merging**